timemachine/ff/handlers/nonbonded.py

```diff
@@ -391,6 +415,22 @@ def static_parameterize(mol):
 
 
 class AM1CCCHandler(SerializableMixIn):
+    """The AM1CCCHandler stands for AM1 Correctable Charge Correction (CCC) which uses OpenEye's AM1 charges[1]
+    along with corrections provided by the Forcefield definition in the form of smirks and charge deltas.
```

Comment:
Could we add a clarification whether the SMIRKS patterns use [Daylight SMARTS syntax](https://www.daylight.com/dayhtml/doc/theory/theory.smarts.html), the [OE SMARTS extension](https://docs.eyesopen.com/toolkits/cpp/oechemtk/SMARTS.html), or the common subset?

(I think we use [OE for matching bond smirks](https://github.com/proteneer/timemachine/blob/2a971ee27cb3b8fc457fee1439af2f8dbd375b72/timemachine/ff/handlers/nonbonded.py#L186) in AM1CCCHandler but [RDKit for matching atom smirks](https://github.com/proteneer/timemachine/blob/2a971ee27cb3b8fc457fee1439af2f8dbd375b72/timemachine/ff/handlers/nonbonded.py#L140) to assign LJ parameters. Not sure if this is just a historical quirk or a necessary aspect of the AM1CCC definition.)

Reply:
> Could we add a clarification whether the SMIRKS patterns use [Daylight SMARTS syntax](https://www.daylight.com/dayhtml/doc/theory/theory.smarts.html), the [OE SMARTS extension](https://docs.eyesopen.com/toolkits/cpp/oechemtk/SMARTS.html), or the common subset?

By this do you mean how we interpret them or how they were defined? 

> (I think we use [OE for matching bond smirks](https://github.com/proteneer/timemachine/blob/2a971ee27cb3b8fc457fee1439af2f8dbd375b72/timemachine/ff/handlers/nonbonded.py#L186) in AM1CCCHandler but [RDKit for matching atom smirks](https://github.com/proteneer/timemachine/blob/2a971ee27cb3b8fc457fee1439af2f8dbd375b72/timemachine/ff/handlers/nonbonded.py#L140) to assign LJ parameters. Not sure if this is just a historical quirk or a necessary aspect of the AM1CCC definition.)

@proteneer can you comment on this? I would be for unifying to a single smirks matching if there is not a good reason for the split.

Reply:
>By this do you mean how we interpret them or how they were defined?

How we want to interpret them.

-----

(Some relevant differences encountered by OpenFF in this context:
(1) for some atomic primitives (`R4`, `R5`, `R6`, ...),  valid symbols in both flavors have intentionally different meanings,
(2) for malformed patterns (patterns containing nonsensical atomic primitives like `r0`, `r1`, `r2` , patterns containing ambiguous trailing bond symbols like `[*:1]~`, etc.), RDKit and OpenEye may have different behavior,
(3) for malformed molecules, RDKit and OpenEye have different aromaticity perception + sanitization behavior + ..., which can cause even simple connectivity primitives like `X3` to have different behavior,
(4) the hybridization primitive `^` is defined in OE flavor but not in Daylight flavor (although note that `^` is defined in RDKit extension of Daylight https://www.rdkit.org/docs/RDKit_Book.html#smarts-support-and-extensions ).

I don't think any of these issues impact the current `_ccc.py` files, but still seems good to be more explicit about which flavor this class assumes, since the constructor can accept an arbitrary smirks list.)

Reply:
> (1) for some atomic primitives (R4, R5, R6, ...), valid symbols in both flavors have intentionally different meanings

Based on reading this (https://docs.eyesopen.com/toolkits/cpp/oechemtk/SMARTS.html), it seems like we would probably prefer the OE toolkits in this case (as it is deterministic), though I don't see any references to `R<n>` in our forcefields.

> (3) for malformed molecules, RDKit and OpenEye have different aromaticity perception + sanitization behavior + ..., which can cause even simple connectivity primitives like X3 to have different behavior

🤯 

It seems to me that it would be best to declare this class as using OE Toolkits SMARTS extension in this PR. And in a separate PR do an investigation of RDKIT and see if they differ.

Reply:
>Based on reading this (https://docs.eyesopen.com/toolkits/cpp/oechemtk/SMARTS.html), it seems like we would probably prefer the OE toolkits in this case (as it is deterministic), though I don't see any references to R<n> in our forcefields.

Agreed about preferring OE behavior in this case, and correct -- no references to R<n> in OpenFF or in our `*_ccc.py` files. (Just this and other handlers could become ambiguous if constructed with an arbitrary `smirks` list in the future.)

>It seems to me that it would be best to declare this class as using OE Toolkits SMARTS extension in this PR. And in a separate PR do an investigation of RDKIT and see if they differ.

That sounds best!